### PR TITLE
Bisrat.demo project

### PR DESCRIPTION
**Pull request by bisrat-walle**

### CODE REVIEW

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	<div style="margin:auto;max-width:600px !important;padding-bottom:30px">
+  <h2><span v-if="id==null">Add Blog</span><span v-if="id!=null">Update Blog</span></h2>
+  
+  <form>
+    <v-text-field
+      v-model="title"
+	  label="Title"
+    ></v-text-field>
+    <v-textarea
+      v-model="content"
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:11]: nit; correct indentation makes the code more readable. you can use `npm run lintfix`

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<template>
+	<div style="margin:auto;max-width:600px !important;padding-bottom:30px">
+  <h2><span v-if="id==null">Add Blog</span><span v-if="id!=null">Update Blog</span></h2>
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:3]: nit; `v-if="id!=null` can be replaced with `v-else`

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  <form>
+    <v-text-field
+      v-model="title"
+	  label="Title"
+    ></v-text-field>
+    <v-textarea
+      v-model="content"
+	  label="Content"
+	  solo
+
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:14]: nit; unnecessary whitespace

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import { mapActions, mapState } from 'vuex';
+
+export default {
+	name: "AddEditComponent",
+    data(){
+        return {
+			id: null,
+			title: "",
+			content: "",
+			titleRules: [v => v.length > 9 || "title is required and must be at least 10 chars"],
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:38]: nit; indentation

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+			contentRules: [v => v.length > 99 || "content is required and must be at least 100 chars"]
+        }
+    },
+	computed: {...mapState("bisrat", ["blog"])},
+    created(){
+        if (this.blog !== null){
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:44]: nit; 
if (this.blog !== null)``` can be replaced with ```if (this.blog)```

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        ...mapActions("bisrat", ["setFormOpen", "createBlog", "updateBlog", "setUpdateBlog", "formOpen"]),
+		onSubmit(){
+			if(this.id == null){
+			  this.createBlog({
+				_id: this.id,
+				title: this.title,
+				content: this.content
+			  });
+			} else {
+			console.log("Updating")
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:60]: nit; no-console

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+			  this.updateBlog({
+				_id: this.id,
+				title: this.title,
+				content: this.content
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:64]: nit; this code is duplicated. it can get refactored it to separate variables.

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+			  });
+			} 
+			this.id = null
+			this.title = ""
+			this.content = ""
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:69]: this can be in separate function.
example 
```
clearFields() {
  this.id = null
  ... 
}
```

File: starter-project-web-vue2/components/bisrat/AddEditBlog.vue [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+			this.setUpdateBlog(null)
+			this.setFormOpen(false)
+		},
+		getButtonText(){
+			if (this.id==null){
+				return "Add"
+			}
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/AddEditBlog.vue:76]: this belongs to computed property instead of method.

File: starter-project-web-vue2/components/bisrat/BlogDetail.vue [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<template>
+	<div style="margin:auto;max-width:600px !important;padding-bottom:30px">
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/BlogDetail.vue:2]: you can use vuetify classes

File: starter-project-web-vue2/components/bisrat/BlogList.vue [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+            color="#385F73"
+            dark
+            class="mb-6"
+			
+          >
+            <v-card-title class="text-h5">
+              {{ blog.title }}
+            </v-card-title>
+
+            <v-card-subtitle> {{ blog.content.substring(0, 10) + " ... " }} </v-card-subtitle>
```
Comment (TripleThreads) [inline starter-project-web-vue2/components/bisrat/BlogList.vue:25]: nit; ` `{{ ${blog.content.substring(0, 10)}... }}` `

File: starter-project-web-vue2/pages/bisrat/blogs/_id.vue [hunk 12]
```diff
@@ -0,0 +1,40 @@
+<template>
+	<div>
+	<div style="display:flex;justify-content:center;padding-bottom:30px">
+		<nuxt-link
+                  style="text-decoration: none; color: inherit"
+                  :to="'/bisrat/'"
```
Comment (TripleThreads) [inline starter-project-web-vue2/pages/bisrat/blogs/_id.vue:6]: nit; `to="bisrat"`

File: starter-project-web-vue2/pages/bisrat/blogs/_id.vue [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+			if (blog._id === id){
+				this.setBlogDetail(blog)
+				break
+			}
+			
+		}
+	},
+	methods: {
+		...mapActions("bisrat", ["setBlogDetail"])
+	}
```
Comment (TripleThreads) [inline starter-project-web-vue2/pages/bisrat/blogs/_id.vue:37]: It can be directly passed through `props` to BlogDetails

File: starter-project-web-vue2/store/bisrat.js [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+	setBlogDetail({ commit }, blog){
+		commit("setBlogDetail", blog)
+	}
+}
+
+export const mutations = {
+    getBlogs: (state, blogs) => {
+		state.blogs = blogs
+		console.log("State.blogs")
+		console.log(state.blogs)
```
Comment (TripleThreads) [inline starter-project-web-vue2/store/bisrat.js:66]: nit; no-console

**Review by TripleThreads — CHANGES_REQUESTED** [review]
Left some comments.

File: starter-project-web-vue2/store/bisrat.js [hunk 15]
```diff
@@ -0,0 +1,93 @@
+import axios from 'axios';
+
+
```
Comment (danielgitk) [inline starter-project-web-vue2/store/bisrat.js:3]: It'd be good if you implemented getters here for the properties you need.
